tests/snuba/search/test_backend.py

```diff
@@ -1217,6 +1229,165 @@ def test_first_release_environments(self):
+        #
+        # groups table:
+        # group    first_release
+        # A        1
+        # B        1
+        # C        2
+        #
+        # groupenvironments table:
+        # group    environment    first_release
+        # A        staging        1
+        # A        production     2
+        #
+        # when querying by first release, the appropriate set of groups should be displayed:
+        #
+        #     first_release: 1
+        #         env=[]: A, B
+        #         env=[production, staging]: A
+        #         env=[staging]: A
+        #         env=[production]: nothing
+        #
+        #     first_release: 2
+        #         env=[]: A, C
+        #         env=[production, staging]: A
+        #         env=[staging]: nothing
+        #         env=[production]: A
+
+        # create an issue/group whose events that occur in 2 distinct environments
+
+        group_a_event_1 = self.store_event(
+            data={
+                'fingerprint': ['group_a'],
+                'event_id': 'aaa' + ('1' * 29),
+                'environment': 'example_staging',
+                'release': 'release_1',
+            },
+            project_id=self.project.id,
+        )
+
+        group_a_event_2 = self.store_event(
+            data={
+                'fingerprint': ['group_a'],
+                'event_id': 'aaa' + ('2' * 29),
+                'environment': 'example_production',
+                'release': 'release_2',
+            },
+            project_id=self.project.id,
+        )
+
+        group_a = group_a_event_1.group
+
+        # get the environments for group_a
+
+        prod_env = group_a_event_2.get_environment()
+        staging_env = group_a_event_1.get_environment()
+
+        # create an issue/group whose event that occur in no environments
+        # but will be tied to release release_1
+
+        group_b_event_1 = self.store_event(
+            data={
+                'fingerprint': ['group_b'],
+                'event_id': 'bbb' + ('1' * 29),
+                'release': 'release_1'
+            },
+            project_id=self.project.id,
+        )
+        assert group_b_event_1.get_environment().name == u''  # has no environment
+
+        group_b = group_b_event_1.group
+
+        # create an issue/group whose event that occur in no environments
+        # but will be tied to release release_2
+
+        group_c_event_1 = self.store_event(
+            data={
+                'fingerprint': ['group_c'],
+                'event_id': 'ccc' + ('1' * 29),
+                'release': 'release_2'
+            },
+            project_id=self.project.id,
+        )
+        assert group_c_event_1.get_environment().name == u''  # has no environment
+
+        group_c = group_c_event_1.group
+
+        # group_a occured in staging environment, and the latest release at the time
+        # of this occurrence was release_1
+
+        GroupEnvironment.get_or_create(
```

Comment:
These `get_or_create` lines don't do anything, you can just remove them. The `GroupEnvironment` rows have already been created previously, so this just fetches them from the db and discards them.

Reply:
Oh interesting. I'll remove this.